src/catalogue/category/cataloguePropertiesForm.component.tsx

```diff
@@ -390,25 +393,111 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
           spacing={1}
           padding={1}
         >
-          <TextField
-            label="Property Name"
-            id={`catalogue-category-form-data-name-${field.cip_placement_id}`}
-            variant="outlined"
-            required={true}
-            value={field.name}
-            onChange={(e) =>
-              handleChange(field.cip_placement_id, 'name', e.target.value)
-            }
-            error={
-              !!catalogueItemPropertyMessage(field.cip_placement_id, 'name')
-            }
-            helperText={
-              catalogueItemPropertyMessage(field.cip_placement_id, 'name')
-                ?.errors?.errorMessage
-            }
-            sx={{ minWidth: '150px' }}
-          />
-          <FormControl sx={{ width: '150px', minWidth: '150px' }}>
+          <FormControl sx={{ width: '200px', minWidth: '200px' }}>
+            <Autocomplete
+              freeSolo
+              options={
+                catalogueItemPropertyTemplates
+                  ? [...catalogueItemPropertyTemplates].sort((a, b) =>
+                      a.name.localeCompare(b.name)
+                    )
+                  : []
+              }
+              getOptionLabel={(option) =>
+                typeof option === 'string' ? option : option.name
+              }
+              value={
+                catalogueItemPropertyTemplates?.find(
+                  (template) => template.name === field.name
+                ) || null
+              }
+              inputValue={field.name}
+              onInputChange={(_event, newValue) => {
+                handleChange(field.cip_placement_id, 'name', newValue);
+              }}
+              onChange={(_event, newValue) => {
+                if (typeof newValue !== 'string' && newValue !== null) {
+                  const updatedFormFields = [...formFields];
```

Comment:
Just a note - I don't believe this prevents updates on the formFields itself as it will only do a shallow copy, not sure if that was the intention though.

Reply:
this should be a separate issue as it used in multiple places to change it to use `JSON.parse(JSON.stringify(array))`

Reply:
#503